### PR TITLE
Tweak printing boxes for unicode binders

### LINKED ISSUE #7765: Unpretty pretty printing of long chains of unicode quantifiers
#### Version

8.8.0

#### Operating system

Linux

#### Description of the problem

Consider the following code:

```
Section test.
Context (very_very_long_type_name1 : Type) (very_very_long_type_name2 : Type).
Context (f : very_very_long_type_name1 -> very_very_long_type_name2 -> Prop).

Lemma test : forall (x : very_very_long_type_name1) (y : very_very_long_type_name2),
    f x y /\ f x y /\ f x y /\ f x y /\ f x y /\ f x y.
Proof.
```
This shows
```
  very_very_long_type_name1 : Type
  very_very_long_type_name2 : Type
  f : very_very_long_type_name1 -> very_very_long_type_name2 -> Prop
  ============================
  forall (x : very_very_long_type_name1) (y : very_very_long_type_name2),
  f x y /\ f x y /\ f x y /\ f x y /\ f x y /\ f x y
```
which I would call pretty readable.

Now let's import the Utf8 notation:
```
From Coq Require Import Utf8.
Section test.
Context (very_very_long_type_name1 : Type) (very_very_long_type_name2 : Type).
Context (f : very_very_long_type_name1 -> very_very_long_type_name2 -> Prop).

Lemma test : forall (x : very_very_long_type_name1) (y : very_very_long_type_name2),
    f x y /\ f x y /\ f x y /\ f x y /\ f x y /\ f x y.
Proof.
```
This shows
```
  ∀ (x : very_very_long_type_name1) (y : very_very_long_type_name2), f x y
                                                                     ∧ f x y
                                                                       ∧ 
                                                                       f x y ∧ 
                                                                       f x y ∧ f x y ∧ f x y
```
The actual goal is crammed in a tight space on the right, making the entire thing pretty hard to read.

Here is a real example from something I am currently working on:
```
  ∀ x : Z, l ↦ #x
           -∗ atomic_acc (⊤ ∖ ⊤) (⊤ ∖ ⊤) (λ (v : val) (q : Qp), l ↦{q} v)
                (l ↦ #x
                 ∗ (AU {{ ∀ v : Z, l ↦ #v }} @ ⊤, ⊤ {{ l ↦ #(v + 1), RET Q -∗ Φ #v }}
                    ={⊤}=∗ AU {{ ∀ v : Z, l ↦ #v }} @ ⊤, ⊤ {{ l ↦ #(v + 1), RET Q -∗ Φ #v }}))
                (λ (v : val) (q : Qp), l ↦{q} v)
                (λ (x' : [tele (_ : val)(_ : Qp)]) (y' : [tele]), l ↦ #x
                                                                  ∗ (AU {{ ∀ 
                                                                        v : Z, 
                                                                        l ↦ #v }} @ ⊤, ⊤
                                                                        {{ l ↦ #(v + 1),
                                                                        RET 
                                                                        Q -∗ Φ #v }}
                                                                     ={⊤}=∗ 
                                                                     (λ (x0 : val) (_ : Qp), 
                                                                      Q
                                                                      -∗ 
                                                                      WP 
                                                                      let: "oldv" := 
                                                                      x0 in
                                                                      if: 
                                                                      CAS #l "oldv" ("oldv" + #1)
                                                                      then "oldv" else 
                                                                      incr #l {{ v, 
                                                                      Φ v }}) x' y'))
```
Notice now the `λ` forces everything after it to be printed in the right half of the screen, making the entire thing very hard to parse.

I am not sure what can be done here, but somehow unicode binders should have their printing boxes set up in a way that the part after the `,` is not indented all the way to the right of all binders.

### PR DESCRIPTION
**Pull request by RalfJung**
**Kind:** bug fix

Fixes / closes #7765

- [x] Added / updated test-suite

### CODE REVIEW

**Comment by herbelin** [comment]
Great.

**Comment by herbelin** [comment]
Noone finds it wrong if I assigned myself to this PR, as far as I'm somehow directly concerned and that the risk of remaining otherwise untreated does not seem neglictible? Then, I would merge it by say Wednesday, leaving some time for further improvements if ever someone has to proposed some.

**Comment by maximedenes** [comment]
This is a notation-related PR more than a library-related PR, so I think it makes perfect sense.

**Comment by RalfJung** [comment]
Thanks!
